io/net/net_data.py

```diff
@@ -85,35 +85,43 @@ def testbigping(self):
         '''
         check with different maximum transfer unit values
         '''
-        msg = "ip addr show  | grep %s | grep -oE '[^ ]+$'" % self.peer
-        cmd = "ssh %s %s" % (self.peer, msg)
+        cmd = "ip addr show  | grep %s | grep -oE '[^ ]+$'" % self.peer
+        cmd = "ssh %s \"%s\"" % (self.peer, cmd)
         errors = []
-        self.peer_interface = process.system_output(cmd, shell=True).strip()
+        peer_interface = process.system_output(cmd, shell=True).strip()
         mtuval = process.system_output("ip link show %s" % self.interface,
                                        shell=True).split()[4]
         for mtu in self.mtu_list:
-            self.log.info("trying with mtu %s" % (mtu))
+            mtu_set = False
+            self.log.info("trying with mtu %s", mtu)
             # ping the peer machine with different maximum transfers unit sizes
             # and finally set maximum transfer unit size to 1500 Bytes
-            msg = "ssh %s \"ip link set %s mtu %s\"" % (self.peer,
-                                                        self.peer_interface,
+            cmd = "ssh %s \"ip link set %s mtu %s\"" % (self.peer,
+                                                        peer_interface,
                                                         mtu)
-            process.system(msg, shell=True)
-            con_msg = "ip link set %s mtu %s" % (self.interface, mtu)
-            process.system(con_msg, shell=True)
+            if process.system(cmd, shell=True, ignore_status=True):
+                self.log.debug("setting mtu value %s in peer failed", mtu)
+                mtu_set = True
+            con_cmd = "ip link set %s mtu %s" % (self.interface, mtu)
+            if process.system(con_cmd, shell=True, ignore_status=True):
+                self.log.debug("setting mtu value %s in host failed", mtu)
+                mtu_set = True
```

Comment:
See? That's the confusion I just mentioned. Here the command failed, so it should not be `True`. `True` here should be attributed in the `else`.